src/Simplify_Stmts.cpp

```diff
@@ -240,7 +240,13 @@ Stmt Simplify::visit(const For *op) {
                                         loop_var_info);
 
         // If we're in the loop, the extent must be greater than 0.
-        ScopedFact fact = scoped_truth(extent_positive);
+        ScopedFact fact_extent_positive = scoped_truth(extent_positive);
+
+        // The loop variable will never exceed the loop bound.
+        Expr loop_var = Variable::make(Int(32), op->name);
+        Expr new_max = mutate(new_min + new_extent, nullptr);
+        ScopedFact fact_loop_var_less_than_extent = scoped_truth(loop_var < new_max);
```

Comment:
Might be worth adding scoped_truth(new_min <= loop_var) while you're here.

Reply:
Good point. My brain is clearly not thinking this through... I committed that as well now.